engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/source/KafkaAvroSourceFactory.scala

```diff
@@ -15,12 +15,10 @@ import pl.touk.nussknacker.engine.avro.schemaregistry.SchemaRegistryProvider
 import pl.touk.nussknacker.engine.avro.typed.AvroSchemaTypeDefinitionExtractor
 import pl.touk.nussknacker.engine.flink.api.process.FlinkSource
 
-import scala.reflect.ClassTag
-
-class KafkaAvroSourceFactory[T: ClassTag](val schemaRegistryProvider: SchemaRegistryProvider[T],
-                                          val processObjectDependencies: ProcessObjectDependencies,
-                                          timestampAssigner: Option[TimestampAssigner[T]])
-  extends BaseKafkaAvroSourceFactory(processObjectDependencies, timestampAssigner) with KafkaAvroBaseTransformer[FlinkSource[T], T]{
+class KafkaAvroSourceFactory(val schemaRegistryProvider: SchemaRegistryProvider,
+                             val processObjectDependencies: ProcessObjectDependencies,
+                             timestampAssigner: Option[TimestampAssigner[Any]])
+  extends BaseKafkaAvroSourceFactory(processObjectDependencies, timestampAssigner) with KafkaAvroBaseTransformer[FlinkSource[Any]]{
```

Comment:
Where are we currently determining type information?

Reply:
`KafkaAvroValueDeserializationSchemaFactory.getProducedType`

Reply:
Good question, it may be not obvious - you have `create[T: ClassTag]` in `KafkaAvroDeserializationSchemaFactory` it will produce type information regards to given `class` and `schemaOpt` - see `ConfluentKafkaAvroDeserializerFactory`. I can add notice in `KafkaAvroDeserializationSchemaFactory` why this `ClassTag` is important.

Reply:
:+1: some comment would be great